curp/src/server/curp_node.rs

```diff
@@ -189,7 +191,9 @@ impl<C: 'static + Command, RC: RoleChange + 'static> CurpNode<C, RC> {
         let id = req.id()?;
         debug!("{} get wait synced request for cmd({id})", self.curp.id());
 
-        let (er, asr) = CommandBoard::wait_for_er_asr(&self.cmd_board, &id).await;
+        let Some((er, asr)) = CommandBoard::wait_for_er_asr(&self.cmd_board, &id).await else {
+            return Ok(WaitSyncedResponse::new_from_result::<C>(None, None)?);
```

Comment:
Why should we consider this case as a result rather than an error?

Reply:
The result `(None, None)` would be converted into `WaitSyncError::Other("can't get er result".to_owned())`. Return `Err()` will be convert into `tonic::Status::internal` error. The `Err()` return will result in a retry in client and continuously fail to obtain results and keep retrying.

 Error handling across RPC looks a bit messy, and some parts don't comply with Rust's conventions. I think the overall error-handling code needs to be refactored.

Reply:
Agree